src/processing/processing.ts

```diff
@@ -158,3 +167,129 @@ export const postProcessApiCallResponse = async (
+ * @param endpoint The endpoint containing processing specifications.
+ * @param apiCallParameters The parameters used in the API call.
+ * @param processingOptions Options to control the async processing behavior like retries and timeouts.
+ *
+ * @returns A promise that resolves to the post-processed API call response.
+ */
+export const postProcessApiCallResponseV2 = async (
+  apiCallResponse: unknown,
+  endpoint: Endpoint,
+  apiCallParameters: ApiCallParameters,
+  processingOptions: GoAsyncOptions = { retries: 0, totalTimeoutMs: DEFAULT_PROCESSING_TIMEOUT_MS }
+): Promise<PostProcessingV2Response> => {
+  const { postProcessingSpecificationV2 } = endpoint;
+  if (!postProcessingSpecificationV2) return { apiCallResponse };
+
+  // We only wrap the code through "go" utils because of the timeout and retry logic. In case of error, the function
+  // just re-throws.
+  const goResult = await go(async () => {
+    const { environment, timeoutMs, value } = postProcessingSpecificationV2;
+    // Provide endpoint parameters without reserved parameters immutably between steps. Recompute them for each
+    // snippet independently because processing snippets can modify the parameters.
+    const endpointParameters = removeReservedParameters(apiCallParameters);
+
+    switch (environment) {
+      case 'Node': {
+        return unsafeEvaluateV2(value, { apiCallResponse, endpointParameters }, timeoutMs);
+      }
+    }
+  }, processingOptions);
+  if (!goResult.success) throw goResult.error;
+
+  return postProcessingV2ResponseSchema.parse(goResult.data);
+};
+
+/**
+ * Post-processes the API call response based on the provided endpoint's processing specifications. Internally it
+ * determines what processing implementation should be used.
+ *
+ * @param apiCallResponse The raw response obtained from the API call.
+ * @param endpoint The endpoint containing processing specifications.
+ * @param apiCallParameters The parameters used in the API call.
+ * @param processingOptions Options to control the async processing behavior like retries and timeouts.
+ *
+ * @returns A promise that resolves to the post-processed API call response.
+ */
+export const postProcessApiCallResponse = async (
+  apiCallResponse: unknown,
+  endpoint: Endpoint,
+  apiCallParameters: ApiCallParameters,
+  processingOptions: GoAsyncOptions = { retries: 0, totalTimeoutMs: DEFAULT_PROCESSING_TIMEOUT_MS }
+): Promise<PostProcessingV2Response> => {
+  const { postProcessingSpecificationV2 } = endpoint;
+  if (postProcessingSpecificationV2) return postProcessApiCallResponseV2(apiCallResponse, endpoint, apiCallParameters);
+
+  const postProcessV1Response = await postProcessApiCallResponseV1(
+    apiCallResponse,
+    endpoint,
+    apiCallParameters,
+    processingOptions
+  );
+  return { apiCallResponse: postProcessV1Response };
+};
+
+/**
+ * Pre-processes API call parameters based on the provided endpoint's processing specifications. Internally it
```

Comment:
Do we want this to happen here? Wouldn't it be better for the caller to decide which implementation they want?

Reply:
I think it's better to do it also here. I initially only had the V1 and V2 function and the caller was forced to handle it. Then I wanted to implement the changes in Airnode and it was quite ugly (basically bunch of unnecessary ifs) so I created this. I'd argue it's actually better this way because the caller doesn't really care which function is called, it just wants the processing result (and in case the caller wants to handle it explicitly, they can).